### PR TITLE
Add the basic details about the current data types

### PR DESCRIPTION
**Pull request by saswatds**

### CODE REVIEW

File: docs/data-types.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+We are assuming most flow programmers would either be coming from a 
+javascript/typescript background, or it might be their first time programming in 
+an API-First language; therefore we have tried to keep the vocabulary of the 
+types close to javascript/typescript.
+
+## Primitive Values
+| type | What it accepts | Example|
+|---------|-----------------|--------|
+|bool     | **true** or **false** | `true`|
```
Comment (samay-v) [inline docs/data-types.md:18]: "Bool" should be "Boolean" right?

File: docs/data-types.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+|string   | utf-8 encoded characters |`foo 😎 bar` |
+|number   | double precision 64-bit values or 64bit unsigned integers| `3.14` <br> `4294967295`|
```
Comment (samay-v) [inline docs/data-types.md:20]: Should there be a floating point int example as well?

File: docs/data-types.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+|timestamp| string containing RFC3339 timestamp| `1985-04-12T23:20:50.52Z`|
+|null     | Exactly one value **null** | `null`
+|regex    | A regular expression | `/ab+c/`|
+|datatype | Symbol of any type | `string` <br> `bool` <br> `null` |
```
Comment (samay-v) [inline docs/data-types.md:24]: Could we say "Symbol of any data type" to avoid confusion

File: docs/data-types.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+|range (pre-proposal) | A range value | `0..10` |
+
+### Boolean
+The boolean value can contain two symbols `true` or `false` as with any programming language. The Boolean type is represented by the symbol `bool`.
+
+The *boolean* value will type-coercion to a *number* or a *string*.
```
Comment (samay-v) [inline docs/data-types.md:30]: Can we say "The *boolean* value can be converted to a *number* or a *string*"

**Review by samay-v — COMMENTED** [review]
Left comments with a few suggested changes

File: docs/data-types.md (hunk 2 above)
Comment (saswatds) [reply docs/data-types.md:20]: `3.14` is floating point and `4294967295` is integer

**Review by samay-v — APPROVED** [review]

**Pull request merged**